src/utils/std.ml

```diff
@@ -40,19 +49,10 @@ module Json = struct
   let list f x =
     `List (List.map f x)
 
-  let print f () x =
-    pretty_to_string (f x)
+  let pretty_to_string : (t -> string) option ref =
+    ref None
```

Comment:
Instead of the option, you could just set it to your default printer as defined above:
```ocaml
fun _ ->
  Printf.sprintf
    "Logger error: `Std.Json.pretty_to_string` \
     is not set. You should initialize that reference with the \
     pretifier of your choice to enable json logging. \
     A common one is `Yojson.Basic.pretty_to_string`."
```